runtime/tchannel_client.go

```diff
@@ -59,66 +59,92 @@ func NewTChannelClient(ch *tchannel.Channel, opt *TChannelClientOption) TChannel
 }
 
 func (c *tchannelClient) writeArgs(call *tchannel.OutboundCall, headers map[string]string, req RWTStruct) error {
-	writer, err := call.Arg2Writer()
+
+	bodyBuf := GetBuffer()
```

Comment:
interesting. Then i can do a better optimization. 

we cannot allocate an arg2 or arg3 writer if the thrift payload does not serialize.